Gems/Blast/Code/Source/Components/BlastFamilyComponent.cpp

```diff
@@ -294,19 +290,38 @@ namespace Blast
         // Spawn the family
         m_family->Spawn(transform);
 
+        BlastFamilyDamageRequestBus::MultiHandler::BusConnect(GetEntityId());
+        BlastFamilyComponentRequestBus::Handler::BusConnect(GetEntityId());
+
         m_isSpawned = true;
     }
 
     void BlastFamilyComponent::Despawn()
     {
         AZ_PROFILE_FUNCTION(Physics);
 
-        m_isSpawned = false;
+        if (!m_isSpawned)
+        {
+            return;
+        }
+
+        // Despawn family before reseting members so all the events are properly called
+        // and all data is available to be cleaned up accordingly.
+        m_family->Despawn();
+
+        // collision handlers should have been cleaned up when the family was despawn
```

Comment:
nit: despawn -> despawned (also above and below the `.` are likely redundant for .cpp comments, just mentioning if you want to remove, not important

Reply:
Thanks, will fix.